src/evaluators/UnaryExpression.js

```diff
@@ -97,22 +101,33 @@ function evaluateDeleteOperation(expr: Value | Reference, realm: Realm) {
 
 function generateRuntimeCall(ref: Reference | Value, ast: BabelNodeUnaryExpression, strictCode: boolean, realm: Realm) {
   invariant(ref instanceof Reference);
-  let baseValue = Environment.GetBase(realm, ref);
-  invariant(baseValue instanceof Value);
-  let propertyName = Environment.GetReferencedName(realm, ref);
-  invariant(typeof propertyName === "string");
+  let value;
+  let propertyName;
 
-  if (!baseValue.isSimpleObject()) {
-    Leak.leakValue(realm, baseValue, ast.loc);
+  // the base reference might be an EnvironmentRecord or a ObjectValue
+  // if it is a EnvironmentRecord, we don't serialize a member expression below
+  if (Environment.IsPropertyReference(realm, ref)) {
+    value = Environment.GetBase(realm, ref);
+    invariant(value instanceof Value);
+    if (!value.isSimpleObject()) {
+      Leak.leakValue(realm, value, ast.loc);
+    }
+    propertyName = Environment.GetReferencedName(realm, ref);
+  } else {
+    value = Environment.GetValue(realm, ref);
+    invariant(value instanceof Value);
+    Leak.leakValue(realm, value, ast.loc);
   }
-  return AbstractValue.createTemporalFromBuildFunction(realm, Value, [baseValue], nodes => {
+
+  return AbstractValue.createTemporalFromBuildFunction(realm, Value, [value], nodes => {
     let arg;
-    if (typeof propertyName === "string") {
+    if (propertyName) {
+      invariant(typeof propertyName === "string");
```

Comment:
Why are you sure that this a string? The type system says it could also be SymbolValue.